sdk/servicebus/azure_messaging_servicebus/src/client.rs

```diff
@@ -0,0 +1,559 @@
+    ///             sub_queue: Some(SubQueue::DeadLetter),
+    ///             ..Default::default()
+    ///         })
+    ///     ).await?;
+    ///     Ok(())
+    /// }
+    /// ```
+    pub async fn create_receiver_for_subscription(
+        &self,
+        topic_name: &str,
+        subscription_name: &str,
+        options: Option<CreateReceiverOptions>,
+    ) -> Result<Receiver> {
+        // For topic subscriptions, the base path is topic/subscriptions/subscription
+        let base_path = format!("{}/subscriptions/{}", topic_name, subscription_name);
+
+        // Build the entity path based on sub_queue option
+        let options = options.unwrap_or_default();
+        let entity_path = if let Some(ref sub_queue) = options.sub_queue {
+            format!("{}{}", base_path, sub_queue.as_path_suffix())
+        } else {
+            base_path
+        };
+
+        self.authorize_path(&entity_path).await?;
+
+        Receiver::new(
+            self.connection.clone(),
+            entity_path,
+            None,
+            options.receive_mode,
+            self.options.clone(),
+        )
+        .await
+    }
+
+    /// Gets the fully qualified namespace.
+    pub fn fully_qualified_namespace(&self) -> &str {
+        &self.namespace
+    }
+
+    /// Authorizes access to a Service Bus entity path using the configured credential.
+    /// This method is used internally by senders and receivers when authentication is required.
+    pub(crate) async fn authorize_path(&self, entity_path: &str) -> azure_core::Result<()> {
+        if let Some(ref authorizer) = self.authorizer {
+            let entity_url = azure_core::http::Url::parse(&format!(
+                "amqps://{}:5671/{}",
+                self.namespace, entity_path
+            ))?;
+
+            authorizer
+                .authorize_path(&self.connection, &entity_url)
+                .await?;
+        }
+        // If no authorizer is configured (e.g., connection string auth), no additional authorization is needed
+        Ok(())
+    }
+
+    /// Closes the client and all associated senders and receivers.
+    pub async fn close(&self) -> Result<()> {
+        self.connection.close().await?;
+        Ok(())
+    }
+}
+
+/// A builder for creating a [`ServiceBusClient`].
+///
+/// This builder is used to create a new [`ServiceBusClient`] with the specified parameters.
+/// It follows the same pattern as other Azure SDK client builders.
+///
+/// # Examples
+///
+/// ```no_run
+/// use azure_messaging_servicebus::ServiceBusClient;
+/// use azure_identity::DeveloperToolsCredential;
+///
+/// #[tokio::main]
+/// async fn main() -> Result<(), Box<dyn std::error::Error>> {
+///     let credential = DeveloperToolsCredential::new(None)?;
+///     let client = ServiceBusClient::builder()
+///         .open("my-servicebus.servicebus.windows.net", credential.clone()).await?;
+///     Ok(())
+/// }
+/// ```
+#[derive(Default)]
+pub struct ServiceBusClientBuilder {
```

Comment:
I believe that guidelines say that builders should be in the `builders` module, but I'm not sure that it matters.